dashboard/test/lib/services/script_seed_test.rb

```diff
@@ -337,6 +339,30 @@ class ScriptSeedTest < ActiveSupport::TestCase
       )
     end
 
+    test 'seed updates lesson programming expressions' do
+      script = create_script_tree
+
+      script_with_changes, json = get_script_and_json_with_change_and_rollback(script) do
+        lesson = script.lessons.first
+        lesson.programming_expressions.first.update!(name: 'newBlock')
+        lesson.programming_expressions.create(
+          key: 'newBlock',
+          name: 'newBlock',
+          programming_environment_id: 1
+        )
+      end
+
+      ScriptSeed.seed_from_json(json)
+      script = Script.with_seed_models.find(script.id)
+
+      assert_script_trees_equal script_with_changes, script
+      lesson = script.lessons.first
+      assert_equal(
+        ['newBlock'],
+        lesson.programming_expressions.map(&:key)
+      )
```

Comment:
I would expect there to be three PEs here, because `num_programming_expressions_per_lesson` is 2